private/pkg/git/cloner.go

```diff
@@ -193,10 +193,18 @@ func (c *cloner) CloneToBucket(
 	// Always checkout the FETCH_HEAD to populate the working directory.
 	// This allows for referencing HEAD in checkouts.
 	buffer.Reset()
+
+	// Include auth when using git-filters, so git can pull the required
+	// objects from the origin.
+	checkoutArgs := []string{"checkout", "--force", "FETCH_HEAD"}
+	if options.Filter != "" {
+		checkoutArgs = append(gitConfigAuthArgs, checkoutArgs...)
+	}
```

Comment:
This approach seems overall reasonable -- we could address this alongside the `sparse-checkout`, for example:

```
if options.Filter != "" {
  checkoutArgs = append(gitConfigAuthArgs, checkoutArgs...)
  if options.SubDir != "" {
    // The sparse checkout of the subdir...
  }
}
```

Reply:
Apologies, I should've commented this earlier -- going through it, it actually makes sense to prepend `gitConfigAuthArgs` to all invocations (since this is how we are fetching). So instead of checking this against `options.Filter != ""`, it should be prepended to all the `checkout` and `sparse-checkout` calls (similar to how we're prepending to the submodules call).